File: RB101_Programming_Foundations/Small Probems/Easy9/Python/grocery_list.py
# Write a method which takes a grocery list (array) of fruits with quantities and converts it into an array of the correct number of each fruit.



'''
Input: array of fruits

Output: expand the array of fruits

Rules


Data structure / Algorithms

    create an empty array for the results
    iterate through the array
        for each subarray, add the fruit * numbers into the result array
    return the result
'''

def buy_fruit(basket):
    result_array = []

    for item in basket:
        result_array +=  [item[0]] * item[1]
    return result_array










print(buy_fruit([["apples", 3], ["orange", 1], ["bananas", 2]]) ==
  ["apples", "apples", "apples", "orange", "bananas","bananas"])

File: RB101_Programming_Foundations/small_problems/Easy9/grocery_list.rb
# input : array of subarray ([fruit, quantity])
# output : simple array of fruits that repeats their respective count numbers

# iterate through the outer array, for each subarray, add push that fruit's name as many times to a storage array with the fruit's counts

def buy_fruit(fruits)
  fruits_bucket = []

  fruits.each do | fruit, quantity|
    quantity.times { fruits_bucket << fruit }
  end

  fruits_bucket
end

puts buy_fruit([["apples", 3], ["orange", 1], ["bananas", 2]]) ==
  ["apples", "apples", "apples", "orange", "bananas","bananas"]